base/setup/usetup/usetup.c

```diff
@@ -4598,14 +4598,19 @@ ProgressCountdown(
                                       24,
                                       TRUE,
                                       FOREGROUND_RED | BACKGROUND_BLUE,
-                                      TimerValue,
+                                      0,
                                       NULL,
                                       MUIGetString(STRING_REBOOTPROGRESSBAR),
                                       ProgressTimeOutStringHandler);
 
+    BarWidth = max(1, ProgressBar->Width);
+    TimerValue = TimeOut * BarWidth;
+    ProgressSetStepCount(ProgressBar, TimerValue);
+
     StartTime = NtGetTickCount();
     CONSOLE_Flush();
 
+    TimerDiv = 1000 / BarWidth;
```

Comment:
Please add just after this line, the following one: `TimerDiv = max(1, TimerDiv);` : this should ensure that the variable will never contain '0'.